diff --git a/ArmPlatformPkg/Library/EblCmdLib/EblCmdLib.c b/ArmPlatformPkg/Library/EblCmdLib/EblCmdLib.c
index 8999910..436aa25 100644
--- a/ArmPlatformPkg/Library/EblCmdLib/EblCmdLib.c
+++ b/ArmPlatformPkg/Library/EblCmdLib/EblCmdLib.c
@@ -297,8 +297,8 @@ EblDumpGcd (
   if (EFI_ERROR (Status)) {
       return Status;
   }
-  AsciiPrint ("    Address Range       Image     Device   Attributes\n");
-  AsciiPrint ("__________________________________________________________\n");
+  AsciiPrint ("    Address Range                       Image     Device   Attributes\n");
+  AsciiPrint ("__________________________________________________________________________\n");
   for (i=0; i < NumberOfDescriptors; i++) {
     AsciiPrint ("MEM %016lx - %016lx",(UINT64)MemorySpaceMap[i].BaseAddress,MemorySpaceMap[i].BaseAddress+MemorySpaceMap[i].Length-1);
     AsciiPrint (" %08x %08x",MemorySpaceMap[i].ImageHandle,MemorySpaceMap[i].DeviceHandle);
@@ -325,13 +325,14 @@ EblDumpGcd (
     if (MemorySpaceMap[i].Attributes & EFI_MEMORY_XP)
         AsciiPrint (" MEM_XP");
 
-    if (MemorySpaceMap[i].GcdMemoryType & EfiGcdMemoryTypeNonExistent)
+//问题单号:DTS2014011706527    修改日期:2014.1.17  修改人:z00201473   问题描述:dumpgcd命令实现有误  
+    if (MemorySpaceMap[i].GcdMemoryType == EfiGcdMemoryTypeNonExistent)
         AsciiPrint (" TYPE_NONEXISTENT");
-    if (MemorySpaceMap[i].GcdMemoryType & EfiGcdMemoryTypeReserved)
+    if (MemorySpaceMap[i].GcdMemoryType == EfiGcdMemoryTypeReserved)
         AsciiPrint (" TYPE_RESERVED");
-    if (MemorySpaceMap[i].GcdMemoryType & EfiGcdMemoryTypeSystemMemory)
+    if (MemorySpaceMap[i].GcdMemoryType == EfiGcdMemoryTypeSystemMemory)
         AsciiPrint (" TYPE_SYSMEM");
-    if (MemorySpaceMap[i].GcdMemoryType & EfiGcdMemoryTypeMemoryMappedIo)
+    if (MemorySpaceMap[i].GcdMemoryType == EfiGcdMemoryTypeMemoryMappedIo)
         AsciiPrint (" TYPE_MEMMAP");
 
     AsciiPrint ("\n");
@@ -347,12 +348,24 @@ EblDumpGcd (
     AsciiPrint ("IO  %08lx - %08lx",IoSpaceMap[i].BaseAddress,IoSpaceMap[i].BaseAddress+IoSpaceMap[i].Length);
     AsciiPrint ("\t%08x %08x",IoSpaceMap[i].ImageHandle,IoSpaceMap[i].DeviceHandle);
 
+    //uniBIOS_y00216284_010_start 2014-12-13 16:17:29
+    //Issue ID:DTS2014121302102 
+    //Description:静态扫描整改
+    /*
     if (IoSpaceMap[i].GcdIoType & EfiGcdMemoryTypeNonExistent)
         AsciiPrint (" TYPE_NONEXISTENT");
     if (IoSpaceMap[i].GcdIoType & EfiGcdMemoryTypeReserved)
         AsciiPrint (" TYPE_RESERVED");
     if (IoSpaceMap[i].GcdIoType & EfiGcdIoTypeIo)
         AsciiPrint (" TYPE_IO");
+    */
+    if (IoSpaceMap[i].GcdIoType == EfiGcdIoTypeNonExistent)
+        AsciiPrint (" TYPE_NONEXISTENT");
+    if (IoSpaceMap[i].GcdIoType == EfiGcdIoTypeReserved)
+        AsciiPrint (" TYPE_RESERVED");
+    if (IoSpaceMap[i].GcdIoType == EfiGcdIoTypeIo)
+        AsciiPrint (" TYPE_IO");
+    //uniBIOS_y00216284_010_end 2014-12-11 15:31:14
 
     AsciiPrint ("\n");
   }
